CONTRIBUTING.md

```diff
@@ -0,0 +1,466 @@
+
+Most documentation can be divided into four categories:
+
+- [Reference](#reference-documentation) - Information-oriented documentation
+
+- [Conceptual](#conceptual-documentation) - Understanding-oriented
+  documentation
+
+- [Procedural](#procedural-documentation)
+
+    - How to - Goal-oriented documentation
+
+    - Tutorial - Learning-oriented documentation
+
+See [Documentation Types](#documentation-types) for more information.
+
+However, comments in your code are very important for maintainability and helping other people
+understand your code. See the [Code Comment Guidelines](#documentation-comments) for style guidelines
+related to comments for your code.
+
+### What documentation style guidelines should I follow?
+
+It is important to follow documentation style guidelines to ensure that the documentation
+created by a large number of contributors can flow together. See
+[Documentation Style Guide](#documentation-style).
+
+### How can I link to other docs in my documentation?
+
+Use absolute paths starting with `/` and ending with `.md`, like [`/getting-started/using-the-substrate-scripts.md`](/substrate/node-template/runtime/src/template.rs).
+Never use relative paths with ".." that point to content outside of `/docs`.
+
+### How can I link to source code in my documentation?
+
+Use the `/rustdocs` to link to specific parts of the Substrate source code, for example a module's `trait`. When referencing specific lines of code, copy those lines into the documentation. However, use this sparingly.
+
+### How can I expose my documentation?
+
+Documentation is only useful when users can find it. Adding links to or from existing documentation
+greatly improves the chances that someone can find your documentation.
+
+Tips for leaving breadcrumbs:
+
+- Table of contents: Add links to documentation in the left side navigation
+  on substrate.dev.
+
+- Top-down linkage: Add links from more general documents to more specific documents to help
+  readers learn more about specific topics. The `conceptual` section is a good
+  starting point for top-down linkage.
+
+- Bottom-up linkage: Add links from more specific documents to more general documents to help
+  readers understand the full context of the topics being discussed. Adding links from
+  module, class, or protocol documentation comments to conceptual documentation overviews can be
+  particularly effective.
+
+- Sideways linkage: Add links to documents on subjects that help readers better understand the
+  content of your document.
+
+---
+
+## Documentation Types
+Documentation is an important part of any product or feature because it lets users know how to
+properly use a feature that has been implemented. These guidelines are meant to be a quick and easy
+reference for types of documentation. For information on documentation style guidelines, see
+[Documentation Style Guide](#documentation-style).
+
+### Conceptual, procedural, or reference documentation
```

Comment:
Maybe this is already in here somewhere, but it should be clear where each of these categories of documentation go. **We need to make the mapping from these categories to the places where we store/create documentation explicit**.  This is pretty subjective but I just feel like this is too easily ignored otherwise. 

Maybe new docs might adhere to a template where the author specifies (1) where they intend to put the docs (2) what categories of docs are covered in their new doc (3) how their new doc communicates/references the other categories of docs. To be clear, I don't like excessive templates or adding unnecessary friction, but I don't want us to ignore the step where we first categorize the new material before finishing a writeup because it should clearly fall into one of these categories and our current workflow isn't oriented around these categories.

Reply:
I don't think process this needs to live in our contribution guidelines. We could create a PR template which does something like this.

Internally, I am a fan of defining ahead of time what kind of doc it is, where it goes, etc...